xarray/core/variable.py

```diff
@@ -1388,6 +1391,12 @@ def reduce(self, func, dim=None, axis=None,
             keep_attrs = _get_keep_attrs(default=False)
         attrs = self._attrs if keep_attrs else None
 
+        if keepdims:
+            for i, d in enumerate(self.dims):
+                if d not in dims:
+                    dims.insert(i, d)
+                    data = np.expand_dims(data, axis=i)
```

Comment:
This works on NumPy arrays, but won't work on dask arrays (or other "duck array" types that we want to support in the future).

Two alternatives that would work here:
1. define `expand_dims` in `duck_array_ops.py` using `_dask_or_eager_func`.
2. expand dimensions by indexing with a tuple built-up out of `np.newaxis`/`slice(None)` instead.

Reply:
Thanks @shoyer, I've added option 2 and a test for dask variables